test/experiment_test.rb

```diff
@@ -565,6 +565,45 @@ def test_switch_respects_time_schedule_even_after_assignment
     end
   end
 
+  def test_is_stop_new_assignments
+    e = Verdict::Experiment.new('test') do
+      groups do
+        group :a, :half
+        group :b, :half
+      end
+      schedule_stop_new_assignment_timestamp Time.new(2020, 1, 15)
+    end
+
+    # new assignments stopped after the stop timestamp
+    Timecop.freeze(Time.new(2020, 1, 16)) do
+      assert e.send(:is_stop_new_assignments?)
+      assert_nil e.switch(1)
+    end
+    # new assignments didn't stop before the stop timestamp
+    Timecop.freeze(Time.new(2020, 1, 3)) do
+      assert !e.send(:is_stop_new_assignments?)
+    end
+  end
+
+  def test_switch_respects_stop_new_assignment_timestamp_even_after_assignment
+    e = Verdict::Experiment.new('test') do
+      groups do
+        group :a, :half
+        group :b, :half
+      end
+    end
+
+    assert_equal :a, e.switch(1)
+
+    e.schedule_stop_new_assignment_timestamp Time.new(2020, 4, 15)
+
+    # switch respects to stop new assignment timestamp even after the assignment occurred
+    Timecop.freeze(Time.new(2020, 4, 16)) do
+      assert e.send(:is_stop_new_assignments?)
+      assert_nil e.switch(1)
```

Comment:
This should have been `:a`. In addition to that, we should assert that a new assignment returns `nil`.